ui/modal/modalRemoveAccount/view.jsx

```diff
@@ -0,0 +1,199 @@
+};
+
+export default function ModalRemoveAccount(props: Props) {
+  const { user,
+    totalBalance,
+    userDeletionSuccess,
+    activeMembershipIds,
+    doHideModal,
+    doUserFetch,
+    doSpendEverything,
+    doUserDeleteAccount,
+    doSendCreditsToOdysee,
+    doClearUserDeletionSuccess,
+    doMembershipCancelForMembershipId,
+  } = props;
+
+  const [isAlreadyPendingDeletion] = React.useState(user.pending_deletion);
+  const [deletionRequestSent, setDeletetionRequestSent] = React.useState(false);
+  const [buttonClicked, setButtonClicked] = React.useState(false);
+  const [isBusy, setIsBusy] = React.useState(false);
+  const [isForfeitChecked, setIsForfeitChecked] = React.useState(false);
+  const [errorOccurred, setErrorOccurred] = React.useState(false);
+
+  const originalActiveMembershipsAmount = React.useRef(activeMembershipIds.length);
+  const hasActiveMemberships = activeMembershipIds.length > 0;
+  const finishedMembershipCancelationsCountRef = React.useRef(0);
+  const [finishedMembershipCancelationsCount, setFinishedMembershipCancelationsCount] = React.useState(finishedMembershipCancelationsCountRef.current);
+  const [someMembershipCancelationFailed, setSomeMembershipCancelationFailed] = React.useState(false);
+
+  const isWalletEmpty = totalBalance <= 0.009;
+  const [forfeitCreditsSuccess, setForfeitCreditsSuccess] = React.useState(isWalletEmpty || undefined);
+  const [cancelingMembershipsSuccess, setCancelingMembershipsSuccess] = React.useState(!hasActiveMemberships || undefined);
+  const preDeletionSuccessChecks = [forfeitCreditsSuccess, cancelingMembershipsSuccess];
+
+  const nothingToWipe = isWalletEmpty && !hasActiveMemberships;
+  const showButton = !buttonClicked && (!isAlreadyPendingDeletion || !nothingToWipe);
+
+  const sendDeletionRequest = React.useCallback(() => {
+    doUserDeleteAccount();
+    setTimeout(doUserFetch, 1000);
+    setDeletetionRequestSent(true);
+    setIsBusy(false);
+  }, [doUserDeleteAccount, doUserFetch]);
+
+  // Tracks if deletion of the account failed
+  React.useEffect(() => {
+    if (userDeletionSuccess === false) {
+      setErrorOccurred(true);
+    }
+  }, [userDeletionSuccess]);
+
+  // Tracks if account is ready for deletion, and triggers the deletion
+  React.useEffect(() => {
+    const waitingForPreDeletionSuccessChecksToFinish = preDeletionSuccessChecks.some((check) => check === undefined);
+    if (waitingForPreDeletionSuccessChecksToFinish) {
+      return;
+    }
+    const somethingFailedWhenPrepairingForDeletion = preDeletionSuccessChecks.some((check) => check === false);
+    if (somethingFailedWhenPrepairingForDeletion) {
+      setErrorOccurred(true);
+      setIsBusy(false);
+    } else if (buttonClicked && !deletionRequestSent && !isAlreadyPendingDeletion) {
+      sendDeletionRequest();
+  }
+  }, [preDeletionSuccessChecks, buttonClicked, deletionRequestSent, isAlreadyPendingDeletion, sendDeletionRequest]);
```

Comment:
Wanted to try do things the "right" way. Not sure where to put it, so made a separate PR https://github.com/OdyseeTeam/odysee-frontend/pull/2944  

If it's good, should we continue with focusing to it from now?

Reply:
I also like the new one, seems like it will be easier to deal with it. 

Should have same behavior than old. (One difference currently is that new one doesn't allow clicking the button on accounts that have active memberships, if account is already pending for deletion. But don't think that's really needed, and current texts didn't/don't mention about memberships anyway.)